src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -2477,8 +2477,72 @@ public IEnumerable<Document> GetRevisionsBinEntries(DocumentsOperationContext co
             }
         }
 
+        public List<(string Id, long Etag)> GetDeletedRevisionsIds(DocumentsOperationContext context, DateTime before,
+            long batchSize, ref long lastEtag, CancellationToken token)
+        {
+            var ids = new List<(string Id, long Etag)>();
+            var count = 0L;
+            var table = new Table(RevisionsSchema, context.Transaction.InnerTransaction);
+            using var _ = GetEtagAsSlice(context, lastEtag, out var startSlice);
+
+            var enumerator = new TransactionForgetAboutStorageIdEnumerator(
```

Comment:
if we pick only 1k now, wonder if the `TransactionForgetAboutStorageIdEnumerator` is still required here

Reply:
Done